src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Snippets.java

```diff
@@ -0,0 +1,489 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.tools.jlink.internal;
+
+import java.lang.classfile.ClassBuilder;
+import static java.lang.classfile.ClassFile.ACC_PUBLIC;
+import static java.lang.classfile.ClassFile.ACC_STATIC;
+import java.lang.classfile.CodeBuilder;
+import java.lang.constant.ClassDesc;
+import static java.lang.constant.ConstantDescs.CD_Integer;
+import static java.lang.constant.ConstantDescs.CD_Object;
+import static java.lang.constant.ConstantDescs.CD_Set;
+import static java.lang.constant.ConstantDescs.CD_int;
+import java.lang.constant.MethodTypeDesc;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.BiConsumer;
+
+import jdk.tools.jlink.internal.Snippets.ElementLoader;
+
+public class Snippets {
+    // Tested page size of string array
+    public static final int STRING_PAGE_SIZE = 8000;
+
+    public static final ElementLoader<String> STRING_LOADER = ElementLoader.of(CodeBuilder::loadConstant);
+    public static final ElementLoader<Integer> INTEGER_LOADER = (cob, value, index) -> {
+        // loadConstant will unbox
+        cob.loadConstant(value)
+        .invokestatic(ClassDesc.ofInternalName("java/lang/Integer"), "valueOf", MethodTypeDesc.of(CD_Integer, CD_int));
+    };
+    public static final ElementLoader<Loadable> LOADABLE_LOADER = (cob, loadable, index) -> loadable.load(cob);
+
+    /**
+     * Describe a reference that can be load onto the operand stack.
+     * For example, an array of string can be described as a Loadable.
+     * The {@link load} method
+     */
+    public sealed interface Loadable {
+        /**
+         * Generate the bytecode to load the Loadable onto the operand stack.
+         * @param cob  The CodeBuilder to add the bytecode for loading
+         */
+        void load(CodeBuilder cob);
+
+        /**
+         * The type of the reference be loaded onto the operatnd stack.
```

Comment:
```suggestion
         * The type of the reference be loaded onto the operand stack.
```

Reply:
Thanks, fixed locally, will wait for more feedback and accumulate fixes with next commit.